book/index.md

```diff
@@ -39,6 +39,55 @@ API resources in the second row (PipelineRun, Snapshot) should primarilly be tho
 data-plane resources. The system responds to user requests by creating and managing the lifecycle of
 these resources.
 
+- An [Application] represents a functionally coherent set of git branches that should be built,
+  tested, and released together. The user provides and names their [Applications]. They are
+  generally long-lived and don't change much after they are created.
+- A [Component] represents an image to build from a particular git branch (and a particular context
+  directory). It is owned by an [Application]. The user provides and names their [Components] and
+  specifies which git repositories and branches they relate to. The user may add, remove, or change
+  [Components] over the lifespan of an [Application].
+- A [Snapshot] represents a collection of particular image builds, mapped to the [Components] of an
+  [Application]. A [Snapshot] is owned by an [Application]. Generally, the [integration-service]
+  produces new [Snapshots] automatically in response to completed builds. It is also possible for
+  the user to construct and provide [Snapshots] to test or release deliberate combinations of
+  images. Old [Snapshots] are garbage collected if not bound to other resources like
+  a [SnapshotEnvironmentBinding] or a [Release].
+- An [IntegrationTestScenario] represents a test that should be run against new [Snapshots] that
+  belong to a particular [Application]. It is owned by an [Application]. The user provides writes
+  tekton pipelines that test their application and registers them with the system by creating
+  [IntegrationTestScenarios].
+- An [Environment] represents a destination that [application-service] and the [gitops-service] can
+  deploy things to. Multiple [Applications] in the same workspace can deploy to the same
```

Comment:
Can we be more specific than "things" here?

Reply:
I just dropped "things", being even less specific.

I'd rather not work on this part, in light of https://github.com/redhat-appstudio/architecture/pull/147